lib/components/WithdrawForm.jsx

```diff
@@ -1,67 +1,182 @@
+    ticketAddress,
+    maxExitFee
+  ]
 
-  const { usersTicketBalance } = usersChainValues || {}
+  await sendTx(
+    setTx,
+    provider,
+    contractAddress,
+    PrizePoolAbi,
+    'withdrawInstantlyFrom',
+    params,
+    'Withdraw'
+  )
+}
 
-  const { burnedCredit, exitFee, timelockDurationSeconds } = exitFees || {}
+const handleWithdrawWithTimelock = async (
+  setTx,
+  provider,
+  contractAddress,
+  ticketAddress,
+  usersAddress,
+  withdrawAmount,
+  decimals
+) => {
+  if (!withdrawAmount) {
+    poolToast.error(`Withdraw Amount needs to be filled in`)
+    return
+  }
 
-  const { withdrawAmount, withdrawType } = vars
+  const params = [usersAddress, ethers.utils.parseUnits(withdrawAmount, decimals), ticketAddress]
 
-  const { setWithdrawAmount, setWithdrawType } = stateSetters
+  await sendTx(
+    setTx,
+    provider,
+    contractAddress,
+    PrizePoolAbi,
+    'withdrawWithTimelockFrom',
+    params,
+    'Withdraw'
+  )
+}
 
-  const handleWithdrawTypeChange = (e) => {
-    setWithdrawType(e.target.value)
-  }
+export const WithdrawForm = (props) => {
+  const { setTx, withdrawAmount, setWithdrawAmount } = props
+
+  const walletContext = useContext(WalletContext)
+  const [poolAddresses] = useAtom(poolAddressesAtom)
+  const [usersAddress] = useAtom(usersAddressAtom)
+  const [poolChainValues] = useAtom(poolChainValuesAtom)
+  const [usersChainValues] = useAtom(userChainValuesAtom)
+  const [network] = useAtom(networkAtom)
 
-  const { isRngRequested, tokenDecimals } = poolChainValues || {}
+  const [exitFees, setExitFees] = useState({
+    earlyExitFee: null,
+    timelockDurationSeconds: null,
+    credit: null,
+    fetched: false
+  })
+  const debouncedWithdrawAmount = useDebounce(withdrawAmount, 300)
 
+  const { prizePool, ticket: ticketAddress } = poolAddresses
+  const provider = walletContext.state.provider
+  const { usersTicketBalance } = usersChainValues
+  const { earlyExitFee, timelockDurationSeconds } = exitFees
+  const { isRngRequested, tokenDecimals, maxTimelockDuration } = poolChainValues
   const poolIsLocked = isRngRequested
-  const tokenSymbol = poolChainValues.tokenSymbol || 'TOKEN'
-  const ticketSymbol = poolChainValues.ticketSymbol || 'TICK'
+  const tokenSymbol = poolChainValues.tokenSymbol
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
 
-  let instantTotal = ethers.utils.bigNumberify(0)
-  if (withdrawAmount) {
-    if (exitFee && exitFee.gt(0) && withdrawType === 'instant') {
-      instantTotal = ethers.utils.parseUnits(withdrawAmount, tokenDecimals).sub(exitFee)
+    if (earlyExitFee.gt(0)) {
+      handleWithdrawInstantly(
+        setTx,
+        provider,
+        prizePool,
+        ticketAddress,
+        usersAddress,
+        withdrawAmount,
+        earlyExitFee,
+        tokenDecimals
+      )
     } else {
-      instantTotal = ethers.utils.parseUnits(withdrawAmount, tokenDecimals)
+      handleWithdrawWithTimelock(
```

Comment:
Something that wasn't mentioned is that we are removing the scheduled / timelock code eventually as it serves no purpose. It would be best to delete all of it from the reference app. 

So we'll still have instant, but instant will be the only withdraw type (ie. we can remove `withdrawType === 'instant'` checks and all scheduled/timelock UI).